Common/UI/DropVisualization/DropTableUIState.cs

```diff
@@ -0,0 +1,321 @@
+﻿#if DEBUG
+using System.Collections.Generic;
+using Microsoft.Xna.Framework.Input;
+using PathOfTerraria.Common.Enums;
+using PathOfTerraria.Common.ItemDropping;
+using PathOfTerraria.Common.UI.Utilities;
+using PathOfTerraria.Core.Items;
+using PathOfTerraria.Core.UI.SmartUI;
+using Terraria.Audio;
+using Terraria.GameContent.UI.Elements;
+using Terraria.GameInput;
+using Terraria.ID;
+using Terraria.Localization;
+using Terraria.ModLoader.UI;
+using Terraria.UI;
+
+namespace PathOfTerraria.Common.UI.DropVisualization;
+
+internal class DropTablePlayer : ModPlayer
+{
+	public static ModKeybind ToggleDropTableUIKey = null;
+
+	public override void Load()
+	{
+		if (Main.dedServ)
+		{
+			return;
+		}
+
+		ToggleDropTableUIKey = KeybindLoader.RegisterKeybind(Mod, "DropTableKey", Keys.OemBackslash);
+	}
+
+	public override void ProcessTriggers(TriggersSet triggersSet)
+	{
+		if (ToggleDropTableUIKey.JustPressed)
+		{
+			SmartUiLoader.GetUiState<DropTableUIState>().Toggle();
+		}
+	}
+}
+
+internal class DropResult(int count)
+{
+	public int Count = count;
+	public bool IsUnique = false;
+	public Dictionary<ItemRarity, int> CountsPerRarity = [];
+
+	public void IncrementRarityCount(ItemRarity rarity)
+	{
+		if (!CountsPerRarity.ContainsKey(rarity))
+		{
+			CountsPerRarity.Add(rarity, 0);
+		}
+
+		CountsPerRarity[rarity]++;
+	}
+}
+
+internal class DropTableUIState : CloseableSmartUi
+{
+	public static readonly Point MainPanelSize = new(920, 550);
+
+	private enum SortMode
+	{
+		None,
+		Count,
+		Alphabetical
+	}
+
+	public override bool IsCentered => true;
+
+	private EditableValueUI _gearRate = null;
+	private EditableValueUI _currencyRate = null;
+	private EditableValueUI _mapRate = null;
+	private EditableValueUI _count = null;
+	private EditableValueUI _level = null;
+	private UIButton<string> _sortButton = null;
+	private UIList _resultList = null;
+	private SortMode _sort = SortMode.None;
+
+	public override int InsertionIndex(List<GameInterfaceLayer> layers)
+	{
+		return layers.FindIndex(x => x.Name == "Vanilla: Hotbar");
+	}
+
+	internal void Toggle()
+	{
+		RemoveAllChildren();
+
+		Width = StyleDimension.Fill;
+		Height = StyleDimension.Fill;
+
+		IsVisible = !IsVisible;
+
+		if (!IsVisible)
+		{
+			return;
+		}
+
+		Main.playerInventory = true;
+
+		CreateMainPanel(false, MainPanelSize, false, true);
+		Panel.VAlign = 0.7f;
+
+		BuildModificationPanel(Panel);
+		BuildDisplay(Panel);
+
+		CloseButton = new UIImageButton(ModContent.Request<Texture2D>($"{PoTMod.ModName}/Assets/UI/CloseButton"));
+		CloseButton.Left.Set(-PointsAndExitPadding - 38, 1);
+		CloseButton.Top.Set(10, 0f);
+		CloseButton.Width.Set(38, 0);
+		CloseButton.Height.Set(38, 0);
+		CloseButton.OnLeftClick += (a, b) =>
+		{
+			Toggle();
+			SoundEngine.PlaySound(SoundID.MenuClose, Main.LocalPlayer.Center);
+		};
+		CloseButton.SetVisibility(1, 1);
```

Comment:
Looks the same as AddCloseButton, unless I'm missing something. 
Perhaps the Toggle(), but maybe that could be added to the base class for less copy-paste?